sources/api/netdog/src/dns.rs

```diff
@@ -0,0 +1,259 @@
+    }
+
+    fn from_config_impl<P>(path: P) -> Result<Option<Self>>
+    where
+        P: AsRef<Path>,
+    {
+        // Ensure we don't attempt to load a nonexistent or empty config file At boot time, the
+        // config file won't exist because it hasn't been generated by the API yet.  After that,
+        // the file will always exist because it's a configuration file for settings in the API and
+        // will always be generated.  If the DNS settings aren't populated, the file will be empty.
+        // We can assume if the file is empty that the settings don't exist.
+        let path = path.as_ref();
+        let config_exists = if Path::exists(path) {
+            let file_len = fs::metadata(path)
+                .context(error::DnsConfMetaSnafu { path })?
+                .len();
+            file_len != 0
+        } else {
+            false
+        };
+
+        if config_exists {
+            let config_str =
+                fs::read_to_string(path).context(error::DnsConfReadFailedSnafu { path })?;
+            let dns_config =
+                toml::from_str(&config_str).context(error::DnsConfParseSnafu { path })?;
+
+            Ok(Some(dns_config))
+        } else {
+            eprintln!("No DNS configuration exists in {}", DNS_CONFIG);
+            Ok(None)
+        }
+    }
+
+    /// Create a DnsSettings from LeaseInfo struct
+    pub(crate) fn from_lease_info(info: &LeaseInfo) -> Self {
+        // Randomize name server order, for libc implementations like musl that send
+        // queries to the first N servers.
+        let mut dns_servers: Vec<IpAddr> = info.dns_servers.clone().into_iter().collect();
+        dns_servers.shuffle(&mut thread_rng());
+
+        Self {
+            nameservers: BTreeSet::from_iter(dns_servers.into_iter()),
+            search: info.dns_search.clone(),
+        }
+    }
+
+    /// Create a DnsSettings using an interface's DHCP lease file
+    pub(crate) fn from_lease_file<P>(path: P) -> Result<Self>
+    where
+        P: AsRef<Path>,
+    {
+        let lease_info = LeaseInfo::from_lease(path).context(error::LeaseParseFailedSnafu)?;
+        Ok(Self::from_lease_info(&lease_info))
+    }
+
+    /// Write resolver configuration for libc.
+    pub(crate) fn write_resolv_conf(&self) -> Result<()> {
+        Self::write_resolv_conf_impl(self, RESOLV_CONF)
+    }
+
+    fn write_resolv_conf_impl<P>(&self, path: P) -> Result<()>
+    where
+        P: AsRef<Path>,
+    {
+        let path = path.as_ref();
+        let mut output = String::new();
+
+        if let Some(s) = &self.search {
+            writeln!(output, "search {}", s.join(" "))
+                .context(error::ResolvConfBuildFailedSnafu)?;
+        }
+
+        for n in &self.nameservers {
+            writeln!(output, "nameserver {}", n).context(error::ResolvConfBuildFailedSnafu)?;
+        }
+
+        fs::write(path, output).context(error::ResolvConfWriteFailedSnafu { path })?;
+        Ok(())
```

Comment:
I think you should be able to just do:

```rust
fs::write(path, output).context(error::ResolvConfWriteFailedSnafu { path })?
```

(Or maybe without the `?`)

Reply:
Good catch - thanks!